### PR TITLE
fix appium image locator

### PR DESCRIPTION
**Pull request by AlmogH**
## Proposed changes
The appium image locator didn't work properly, appium expects a base64 string of the image and not the image location.
I changed the using.value and changed the test accordingly.

this is my first pull request if there is anything else I should do let me know and I will change it

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/webdriverio/tests/findStrategy.test.js [hunk 1]
```diff
@@ -379,21 +380,19 @@ describe('selector strategies helper', () => {
     })
 
     it('should find an mobile element using image string', () => {
+        jest.mock('fs')
```
Comment (christian-bromann) [inline packages/webdriverio/tests/findStrategy.test.js:383]: This should be moved to the top of the file.

File: packages/webdriverio/tests/findStrategy.test.js [hunk 2]
```diff
@@ -379,21 +380,19 @@ describe('selector strategies helper', () => {
...
+        fs.readFileSync = jest.fn()
```
Comment (christian-bromann) [inline packages/webdriverio/tests/findStrategy.test.js:384]: no need for this one

**Review by christian-bromann — CHANGES_REQUESTED** [review]
I think the unit tests need to be updated too as we should also check of the value is no a base64 string

**Comment by AlmogH** [comment]
> 
> 
> I think the unit tests need to be updated too as we should also check of the value is no a base64 string

I'm not sure what do you mean, how to change the unit tests? can you please elaborate

**Comment by christian-bromann** [comment]
> I'm not sure what do you mean, how to change the unit tests? can you please elaborate

Sorry for not being clear. You removed the value checks, e.g. `expect(element.value).toBe('/test.jpg')`. In order to ensure that we send proper base64 data to Appium I suggest to setup the `readFileSync` so that it returns a string that we can then check to be a valid base64 string.

**Comment by AlmogH** [comment]
> 
> 
> > I'm not sure what do you mean, how to change the unit tests? can you please elaborate
> 
> Sorry for not being clear. You removed the value checks, e.g. `expect(element.value).toBe('/test.jpg')`. In order to ensure that we send proper base64 data to Appium I suggest to setup the `readFileSync` so that it returns a string that we can then check to be a valid base64 string.

Sorry I'm still not quite sure what is it you want me to change, should I mock `readFileSync` in a way that it returns a valid base64 and an invalid base64 or add real images? or something else completely

**Comment by christian-bromann** [comment]
If you define `jest.mock('fs')` all `fs` methods are already mocked. Please add for `readFileSync` a return value (some random non base64 string) so that we can check the value property to be a proper base64 string.

**Comment by AlmogH** [comment]
`readFileSync`  is a `fs`  method, so should I do something like this: `fs.readFileSync = jest.fn(()=>'random string')`.

right now the value returned by findStrategy is undefined thanks to the `jest.mock('fs')`

**Comment by christian-bromann** [comment]
> `readFileSync` is a `fs` method, so should I do something like this: `fs.readFileSync = jest.fn(()=>'random string')`.

Yes

**Comment by christian-bromann** [comment]
@AlmogH sorry for not getting back to you earlier. Would you mind to rebase on top of current master?

**Comment by DeepTom** [comment]
@AlmogH 
After this PR completion, would $('/image.png'); work ?
where image.png is a base64 encoded file. Or do we have to provide something else to the $() method ?

**Comment by christian-bromann** [comment]
I think something went wrong with the rebase 🤔

**Comment by AlmogH** [comment]
@DeepTom you will need to give the full path of the image so something like this `$(__dirname + '../image.png')`
@christian-bromann First time I did rebase, followed this [guide](https://medium.com/@topspinj/how-to-git-rebase-into-a-forked-repo-c9f05e821c8a) what should I change?

**Comment by christian-bromann** [comment]
@AlmogH can you check out latest master and create a new branch. Then cherry-pick your commits into that branch:

- 00e7a25
- 19b68ee
- 461b957

Then just force push on it. I also see that you started working on `master` in your fork rather than creating a branch for it as well. I think a different way to fix this is:

```sh
# create a new branch
git checkout -b feature-branch
# remove master branch
git branch -D master
# check out origin master
git checkout origin/master
# create a new local master branch
git checkout -b master
# update feature branch
git checkout feature-branch
# rebase on top of it
git rebase master
```

**Comment by AlmogH** [comment]
@christian-bromann I hope that now it is ok

File: packages/webdriverio/tests/findStrategy.test.js [hunk 3]
```diff
@@ -379,23 +381,25 @@ describe('selector strategies helper', () => {
     })
 
     it('should find an mobile element using image string', () => {
+        fs.readFileSync = jest.fn(()=>'random string')
```
Comment (christian-bromann) [inline packages/webdriverio/tests/findStrategy.test.js:384]:
```suggestion
        fs.readFileSync = jest.fn(() => 'random string')
```

File: packages/webdriverio/tests/findStrategy.test.js [hunk 4]
```diff
@@ -1,4 +1,6 @@
 import { findStrategy } from '../src/utils/findStrategy'
+import fs from 'fs'
+jest.mock('fs')
```
Comment (christian-bromann) [inline packages/webdriverio/tests/findStrategy.test.js:3]:
```suggestion

jest.mock('fs')
```

File: packages/webdriverio/src/utils/findStrategy.js [hunk 5]
```diff
@@ -1,5 +1,6 @@
 import { W3C_SELECTOR_STRATEGIES } from '../constants'
 import isPlainObject from 'lodash.isplainobject'
+import fs from 'fs'
```
Comment (christian-bromann) [inline packages/webdriverio/src/utils/findStrategy.js:3]:
```suggestion
import fs from 'fs'
import isPlainObject from 'lodash.isplainobject'

import { W3C_SELECTOR_STRATEGIES } from '../constants'
```

**Review by christian-bromann — COMMENTED** [review]
some minor formatting

**Review by christian-bromann — APPROVED** [review]
👍

**Pull request merged**